services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java

```diff
@@ -880,6 +880,8 @@ final protected BrokeredIdentityContext validateJwt(EventBuilder event, String s
             return context;
         } catch (IOException e) {
             logger.debug("Unable to extract identity from identity token", e);
+            event.error(Errors.INVALID_TOKEN);
+            event.detail(Details.REASON, "Unable to extract identity from identity token: " + e.getMessage());
```

Comment:
One note: When you call `event.detail` after `event.error`, you might be set the "detail" too late? As `event.error` also sends the event to the event listener. I guess this is not intended behaviour?

Reply:
Thanks, I misunderstood the API. Didn't know this method was different from the others. Reworked.